### PR TITLE
Feature/scripts backend

### PR DESCRIPTION
**Pull request by NoSurprises**

### CODE REVIEW

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt [hunk 1]
```diff
@@ -122,4 +105,24 @@ class HomeRepositoryImpl : HomeRepository, DeviceChannelOutput, RemoteStorageInp
         localStorage.removeDevice(device)
         remoteStorage.removeDevice(device)
     }
+
+
+    override suspend fun findController(guid: Long): BaseController {
```
Comment (ZheMadScientist) [inline raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt]: override?

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt [hunk 2]
```diff
@@ -122,4 +105,24 @@ class HomeRepositoryImpl : HomeRepository, DeviceChannelOutput, RemoteStorageInp
...
+        val devices = localStorage.getDevices()
+
+        for (device in devices) {
+            return device.controllers.find { it.guid == guid } ?: continue
+        }
+        throw NoControllerException()
+    }
+
+    override suspend fun findDevice(controller: BaseController): IotDevice {
```
Comment (ZheMadScientist) [inline raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt]: same

File: raspberry/SmartHome/app/src/test/java/smarthome/raspberry/data/HomeRepositoryImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        runBlocking {
+            val repo: DeviceChannelOutput =
+                    HomeRepositoryImpl(localStorage, remoteStorage, mapOf(
+                            device_A.javaClass to deviceChannel_A
+                    ))
+
+            whenever(localStorage.getDevices()).thenReturn(mutableListOf())
+
+            shouldThrow<NoControllerException> {
+                repo.findController(123)
```
Comment (ZheMadScientist) [inline raspberry/SmartHome/app/src/test/java/smarthome/raspberry/data/HomeRepositoryImplTest.kt]: DeviceChannelOutput does not have findController function

File: raspberry/SmartHome/arduinodevices/src/test/java/smarthome/raspberry/arduinodevices/server/WebServerTest.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    private val session: NanoHTTPD.IHTTPSession = mock()
+    private val deviceChannelOutput: DeviceChannelOutput = mock()
+    private val handler = WebHandler(deviceChannelOutput)
+    private val controller = BaseController("")
+
+    @Before
+    fun setUp() {
+        runBlocking {
+            whenever(deviceChannelOutput.findController(any())).thenReturn(controller)
```
Comment (ZheMadScientist) [inline raspberry/SmartHome/arduinodevices/src/test/java/smarthome/raspberry/arduinodevices/server/WebServerTest.kt]: DeviceChannelOutput does not have findController function

File: raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt [hunk 5]
```diff
@@ -18,10 +14,9 @@ internal abstract class BaseRequestHandler : RequestHandler {
         return NanoHTTPD.Response(NanoHTTPD.Response.Status.BAD_REQUEST, MIME_PLAINTEXT, "Invalid request: $message")
     }
 
-    protected fun getController(params: Map<String, String>): BaseController {
-        // todo add checks so it won't crash
+    protected suspend fun getController(params: Map<String, String>): BaseController {
         val controllerGuid = params.getValue("controller_guid").toLong()
-        return input.findController(controllerGuid)
+        return output.findController(controllerGuid)
```
Comment (ZheMadScientist) [inline raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt]: DeviceChannelOutput does not have findController function

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt (hunk 1 above)
Comment (NoSurprises) [reply raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt]: override, whats the question?

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt (hunk 2 above)
Comment (NoSurprises) [reply raspberry/SmartHome/app/src/main/java/smarthome/raspberry/data/HomeRepositoryImpl.kt]: same. override, whats the question?

File: raspberry/SmartHome/app/src/test/java/smarthome/raspberry/data/HomeRepositoryImplTest.kt (hunk 3 above)
Comment (NoSurprises) [reply raspberry/SmartHome/app/src/test/java/smarthome/raspberry/data/HomeRepositoryImplTest.kt]: why? 
`interface DeviceChannelOutput {
    suspend fun onNewDevice(device: IotDevice)
    suspend fun onNewState(controller: BaseController)
    suspend fun findController(guid: Long): BaseController
    suspend fun findDevice(controller: BaseController): IotDevice
}`

File: raspberry/SmartHome/arduinodevices/src/test/java/smarthome/raspberry/arduinodevices/server/WebServerTest.kt (hunk 4 above)
Comment (NoSurprises) [reply raspberry/SmartHome/arduinodevices/src/test/java/smarthome/raspberry/arduinodevices/server/WebServerTest.kt]: it does

File: raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt (hunk 5 above)
Comment (NoSurprises) [reply raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/server/httphandlers/BaseRequestHandler.kt]: it does

**Pull request merged**